.github/workflows/ci-cd.yml

```
@@ -1,249 +1,410 @@
+          CI_ENVIRONMENT: "test"
+        run: |
+          chmod +x scripts/ci-cd-integration.sh
+          scripts/ci-cd-integration.sh test
+
+      - name: Coverage Check (non-blocking)
+        run: |
+          make coverage-check
+        continue-on-error: true
+
+      - name: Upload test results
+        uses: actions/upload-artifact@v3
```

actionlint (1.7.7)

the runner of "actions/upload-artifact@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)